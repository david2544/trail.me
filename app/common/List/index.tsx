import React from 'react';
import classnames from 'classnames';
import styles from './styles.module.scss';

interface ListProps {
  items: { id: string; content: JSX.Element | string }[];
  listClassName?: string;
}

const List: React.FC<ListProps> = ({ items, listClassName }: ListProps) => (
  <ul className={classnames(listClassName, styles.list)}>
    {items.map(item => (
      <li className={classnames(styles.listItem)} key={item.id}>
        <span>{item.content}</span>
      </li>
    ))}
  </ul>
);

export default List;
