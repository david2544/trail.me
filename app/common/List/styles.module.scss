
.list {
  list-style-type: none;
  text-align: left;
  padding-left: 0px;
  margin-top: 0px;
  margin-bottom: 11px;

  .listItem {
    position: relative;
    padding-bottom: 15px;

    &:last-child {
      padding-bottom: 0;
    }

    .icon {
      margin-right: 10px;
    }
  }

  .defaultIcon {
    padding-left: 30px;

    &:before {
      content: '';
      display: block;
      position: absolute;
      top: 7px;
      left: 0;
      border: 5px solid transparent;
      border-right-color: white;
      border-bottom-color: white;
    }
  }
}

.negativeMargin {
  margin-left: -30px;
}
