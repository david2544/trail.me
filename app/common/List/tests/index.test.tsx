import React from 'react';
import { shallow } from 'enzyme';
import List from '../index';

describe('<List />', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<List items={[{ id: '1', content: 'some content' }]} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapshot when icon is provided', () => {
    const wrapper = shallow(<List items={[{ id: '1', content: 'some content' }]} />);

    expect(wrapper).toMatchSnapshot();
  });
});
