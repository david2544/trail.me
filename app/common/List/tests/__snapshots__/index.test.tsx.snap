// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> should match snapshot 1`] = `
<ul
  className="list negativeMargin"
>
  <li
    className="listItem defaultIcon"
    key="1"
  >
    <span>
      some content
    </span>
  </li>
</ul>
`;

exports[`<List /> should match snapshot when icon is provided 1`] = `
<ul
  className="list"
>
  <li
    className="listItem"
    key="1"
  >
    <FontAwesomeIcon
      border={false}
      className="icon"
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            496,
            512,
            Array [],
            "f118",
            "M248 8C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm80 168c17.7 0 32 14.3 32 32s-14.3 32-32 32-32-14.3-32-32 14.3-32 32-32zm-160 0c17.7 0 32 14.3 32 32s-14.3 32-32 32-32-14.3-32-32 14.3-32 32-32zm194.8 170.2C334.3 380.4 292.5 400 248 400s-86.3-19.6-114.8-53.8c-13.6-16.3 11-36.7 24.6-20.5 22.4 26.9 55.2 42.2 90.2 42.2s67.8-15.4 90.2-42.2c13.4-16.2 38.1 4.2 24.6 20.5z",
          ],
          "iconName": "smile",
          "prefix": "fas",
        }
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size="lg"
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
    <span>
      some content
    </span>
  </li>
</ul>
`;
